### PR TITLE
Support for TF 2 estimator

### PR DESCRIPTION
**Pull request by vandanavk**
### Description of changes:
Support for keras to estimator and EstimatorHook

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/tensorflow2/test_estimator.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def test_estimator(out_dir, tf_eager_mode, saveall):
     tnames = trial.tensor_names()
     assert len(trial.steps()) > 0
     if saveall:
-        assert len(tnames) == 301
+        assert len(tnames) >= 301
```
Comment (NihalHarish) [inline tests/tensorflow2/test_estimator.py]: Comment.

File: tests/tensorflow2/test_estimator.py [hunk 2]
```diff
@@ -56,6 +56,6 @@ def test_linear_classifier(out_dir, tf_eager_mode, saveall):
     tnames = trial.tensor_names()
     assert len(trial.steps()) > 0
     if saveall:
-        assert len(tnames) == 224
+        assert len(tnames) >= 224
```
Comment (NihalHarish) [inline tests/tensorflow2/test_estimator.py]: Comment

**Comment by vandanavk** [comment]
> Can you write an example notebook in sagemaker-examples ?

Sure.

File: tests/tensorflow2/test_estimator.py (hunk 1 above)
Comment (NihalHarish) [reply tests/tensorflow2/test_estimator.py]: Change the assert statements to something more meaningful and add comments explaining what 301 means

**Pull request merged**